nsiqcppstyle_checker.py

```diff
@@ -344,7 +344,12 @@ def __init__(self, filename, data=None):
         self.data = data
         if data is None:
             f = open(filename)
```

Comment:
Let's use `with` here since the file should be closed after reading it

Reply:
Always a good thing - modified several other places in similar fashion.

Done.